ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -398,61 +401,506 @@ void HttpOperation::Cleanup()
   }
 }
 
+#if LIBCURL_VERSION_NUM >= 0x072200
+// CURL 7.34.0 (0x07 0x22 0x00)
+// CURL_SSLVERSION_TLSv1_0
+// CURL_SSLVERSION_TLSv1_1
+// CURL_SSLVERSION_TLSv1_2
+#  define HAVE_TLS_10_to_12
+#endif
+
+#if LIBCURL_VERSION_NUM >= 0x073400
+// CURL 7.52.0 (0x07 0x34 0x00)
+// CURL_SSLVERSION_TLSv1_3;
+#  define HAVE_TLS_13
+#endif
+
+static long parse_min_ssl_version(std::string version)
+{
+#ifdef HAVE_TLS_10_to_12
+  if (version == "1.0")
+  {
+    return CURL_SSLVERSION_TLSv1_0;
+  }
+
+  if (version == "1.1")
+  {
+    return CURL_SSLVERSION_TLSv1_1;
+  }
+
+  if (version == "1.2")
+  {
+    return CURL_SSLVERSION_TLSv1_2;
+  }
+#endif
+
+#ifdef HAVE_TLS_13
+  if (version == "1.3")
+  {
+    return CURL_SSLVERSION_TLSv1_3;
+  }
+#endif
+
+  return 0;
+}
+
+static long parse_max_ssl_version(std::string version)
+{
+#ifdef HAVE_TLS_10_to_12
+  if (version == "1.0")
+  {
+    return CURL_SSLVERSION_MAX_TLSv1_0;
+  }
+
+  if (version == "1.1")
+  {
+    return CURL_SSLVERSION_MAX_TLSv1_1;
+  }
+
+  if (version == "1.2")
+  {
+    return CURL_SSLVERSION_MAX_TLSv1_2;
+  }
+#endif
+
+#ifdef HAVE_TLS_13
+  if (version == "1.3")
+  {
+    return CURL_SSLVERSION_MAX_TLSv1_3;
+  }
+#endif
+
+  return 0;
+}
+
+const char *HttpOperation::GetCurlErrorMessage(CURLcode code)
+{
+  const char *message;
+
+  if (curl_error_message_[0] != '\0')
+  {
+    // Will contain contextual data
+    message = curl_error_message_;
+  }
+  else
+  {
+    // Generic message
+    message = curl_easy_strerror(code);
+  }
+
+  return message;
+}
+
+template <class T>
+CURLcode HttpOperation::SetCurlOption(CURLoption option, T value)
+{
+  CURLcode rc;
+
+  rc = curl_easy_setopt(curl_resource_.easy_handle, option, value);
+
+  if (rc != CURLE_OK)
+  {
+    const char *message = GetCurlErrorMessage(rc);
+    OTEL_INTERNAL_LOG_ERROR("CURL, set option <" << std::to_string(option) << "> failed: <"
+                                                 << message << ">");
+  }
+
+  return rc;
+}
+
 CURLcode HttpOperation::Setup()
 {
   if (!curl_resource_.easy_handle)
   {
     return CURLE_FAILED_INIT;
   }
 
-  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_VERBOSE, 0);
+  CURLcode rc;
+
+  curl_error_message_[0] = '\0';
+  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_ERRORBUFFER, curl_error_message_);
+
+  rc = SetCurlOption(CURLOPT_VERBOSE, 0);
```

Comment:
Should it be `SetCurlOption(CURLOPT_VERBOSE, 0L)` ?

Reply:
Fixed to SetCurlLongOption().